### PR TITLE
migrate monitoring controller resource handling to the dynamic client

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Migrate monitoring controller resource handling to the dynamic client

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/resources/prometheus/role.go [hunk 1]
```diff
@@ -4,42 +4,36 @@ import (
...
-func Role(data resources.RoleDataProvider, existing *rbacv1.Role) (*rbacv1.Role, error) {
-	var r *rbacv1.Role
-	if existing != nil {
-		r = existing
-	} else {
-		r = &rbacv1.Role{}
-	}
-
-	r.Name = resources.PrometheusRoleName
-	r.OwnerReferences = []metav1.OwnerReference{data.GetClusterRef()}
```
Comment (nikhita) [inline api/pkg/resources/prometheus/role.go:20]: Don't we need this too? :thinking:

File: api/pkg/resources/ensure.go [hunk 2]
```diff
@@ -20,80 +17,6 @@ import (
 
 //go:generate go run ../../codegen/reconcile/main.go
 
-// EnsureRole will create the role with the passed create function & create or update it if necessary.
-// To check if it's necessary it will do a lookup of the resource at the lister & compare the existing Role with the created one
-func EnsureRole(data RoleDataProvider, create RoleCreatorDeprecated, lister rbacv1lister.RoleNamespaceLister, client rbacv1client.RoleInterface) error {
```
Comment (nikhita) [inline api/pkg/resources/ensure.go:25]: :tada:

File: api/pkg/resources/prometheus/role.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/resources/prometheus/role.go:20]: The owner ref gets set via a wrapper: https://github.com/kubermatic/kubermatic/pull/3036/files#diff-ecf2aee6f50290e259cb8f1f83f27116R56

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

/assign @xrstf 
for approval

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**